xarray/backends/zarr.py

```diff
@@ -1011,6 +1060,10 @@ def _open_existing_array(self, *, name) -> ZarrArray:
         else:
             zarr_array = self.zarr_group[name]
 
+        # update the model of the underlying zarr group
```

Comment:
Hmm... I guess this is needed in the case we create a new array?

Reply:
yes, the `ZarrStore` can create or update arrays, will will invalidate any caching unless we track those additions

Reply:
I'm wondering if we actually need those new array names explicitly somewhere. Feels cleaner if `members` simply tracked the state of the group when it was opened, and before any modifications by the current call to `to_zarr`.

Reply:
it does _feel_ cleaner, but iirc I had tests failing before I added the cache updates, because the tests re-used `ZarrStore` instances after mutating the underlying zarr group. we would need to change those tests if we want to remove the cache updates

Reply:
These should be only those tests that use `create_store` where we should opt-out of caching by default.

let's discuss on the call

Reply:
updating the cache is now removed